back/app/controllers/web_api/v1/ideas_controller.rb

```diff
@@ -233,6 +246,19 @@ def serialization_options
       }
     end
   end
+
+  def mark_custom_field_values_to_clear!
+    # We need to explicitly mark which custom field values
+    # should be cleared so we can distinguish those from
+    # the custom field value updates cleared out by the
+    # policy (which should stay like before instead of
+    # being cleared out).
+    if @idea&.custom_field_values && params[:idea][:custom_field_values]
```

Comment:
Consider using an early return here. It makes the method shorter, it reduces indentation levels so that the method is easier to read, and it makes clear under what conditions nothing has to be cleared.
```suggestion
    return unless @idea&.custom_field_values && params[:idea][:custom_field_values]
```
The conditions can be improved, because an empty array is a truthy  value:

```suggestion
    return unless @idea&.custom_field_values.present? && params[:idea][:custom_field_values].present?
```